test/datasource.test.js

```diff
@@ -155,6 +155,171 @@ describe('DataSource', function() {
     dataSource.connector.should.equal(mockConnector);
   });
 
+  it('should set states correctly with eager connect', function(done) {
+    var mockConnector = {
+      name: 'loopback-connector-mock',
+      initialize: function(ds, cb) {
+        ds.connector = mockConnector;
+        this.connect(cb);
+      },
+
+      connect: function(cb) {
+        process.nextTick(function() {
+          cb(null);
+        });
+      },
+    };
+    var dataSource = new DataSource(mockConnector);
+    // DataSource is instantiated
+    // connected: false, connecting: false, initialized: false
```

Comment:
I think the comments showing the status i.e. `// connected: false, connecting: false, initialized: false` can be removed since the assertions below them show what we expect.